src/app/file-manager/file-manager.module.ts

```diff
@@ -62,6 +68,17 @@ import { MatDialog } from '@angular/material';
     StatusService,
     FilesDatabase,
     FilesService,
+    {
+      provide: LoginService,
+      deps: [ElectronIpcService, Store, FirecloudService],
+      useFactory(electronIpc: ElectronIpcService, store: Store<any>, firecloudService: FirecloudService) {
+        if (environment.testing) {
```

Comment:
Is there no way to avoid doing this every place the service is used? I would expect that there would be a way to do this in the service definition code, not at the point were the service is imported. This way seems like you'd end up repeating this test in different modules.

Reply:
Yes, that's correct. In fact we don't need to do this any more at least for the tests.